### ISSUE TITLE
CrossEntropyLoss

### ISSUE THREAD

**Issue by jy1993** [pytorch/pytorch#12299]
Code:
import numpy as np
import torch
from sklearn.metrics import log_loss
from torch import nn

y_pred = np.array([[0.2, 0.3, 0.5], [0.2, 0.3, 0.5], [0.2, 0.3, 0.5]])
y_true = [0, 1, 2]
sklearn_loss = log_loss(y_true, y_pred)
loss_fun = nn.CrossEntropyLoss()
pytorch_loss = loss_fun(torch.Tensor(y_pred), torch.Tensor(y_true).long())
loss = -(np.log(0.2) + np.log(0.3) + np.log(0.5))/3
print(f'sklearn log loss : {sklearn_loss}, pytorch loss: {pytorch_loss}, loss: {loss}')

Result:
sklearn log loss : 1.168852632439994, pytorch loss: 1.1064976453781128, loss: 1.168852632439994

The CrossEntropyLoss gives a log_loss differs from the one calculated by sklearn. Is there something wrong?

Pytorch Version: 0.4.1
Windows 10

**Comment by SsnL** [comment]
The losses you calculated via sklearn and numpy are just nll loss. Cross entropy loss has a log sofmax step before nll.
